tests/org.eclipse.ui.editors.tests/src/org/eclipse/ui/internal/texteditor/stickyscroll/StickyScrollingHandlerTest.java

```diff
@@ -129,13 +135,17 @@ public void testPreferencesUpdated() {
 	@Test
 	public void testThrottledExecution() throws InterruptedException {
 		when(linesProvider.get(100, sourceViewer)).thenReturn(List.of(new StickyLine("line 10", 9)));
+		when(linesProvider.get(200, sourceViewer)).thenReturn(List.of(new StickyLine("line 10", 9)));
+		when(linesProvider.get(300, sourceViewer)).thenReturn(List.of(new StickyLine("line 10", 9)));
+		when(linesProvider.get(400, sourceViewer)).thenReturn(List.of(new StickyLine("line 10", 9)));
 
 		stickyScrollingHandler.viewportChanged(100);
 		stickyScrollingHandler.viewportChanged(200);
 		stickyScrollingHandler.viewportChanged(300);
+		waitInUi(20);
```

Comment:
this looks odd. Why only wait on the second-to-last viewport change?

Reply:
Yes, you are right. I added a delay between every call to have more realistic scenario. Thanks.